File: commands/admin/join.js
/* global client guild */

module.exports = {
  name: 'join',
  description: 'Simulerar att en medlem joinar',
  longDescription: 'Simulerar att en medlem joinar',
  aliases: [],
  group: 'admin',
  usage: 'join <(@user)>',
  example: 'join @Ippyson#6200',
  serverOnly: true,
  adminOnly: true,
  async execute( message, args ) {
    let user = await message.mentions.members.first();
    if ( !user ) user = await guild.fetchMember( message.author );

    return client.emit( 'guildMemberAdd', user );
  },
};

File: commands/admin/join.go
package admin

import (
	"github.com/bwmarrin/discordgo"
	"github.com/digitalungdom-se/dub/events"
	"github.com/digitalungdom-se/dub/internal"
	"github.com/digitalungdom-se/dub/pkg"
)

var Join = internal.Command{
	Name:        "join",
	Description: "Simulerar en användare joinar",
	Aliases:     []string{},
	Group:       "admin",
	Usage:       "join @<user>",
	Example:     "join @kelszo#6200",
	ServerOnly:  true,
	AdminOnly:   true,

	Execute: func(ctx *pkg.Context, server *internal.Server) error {
		var member *discordgo.Member

		if len(ctx.Message.Mentions) == 0 {
			ctx.Reply("Du måste ange en användare som ska joina.")
			return nil
		}

		for _, guildMember := range server.Guild.Members {
			if guildMember.User.ID == ctx.Message.Mentions[0].ID {
				member = guildMember
			}
		}

		guildMemberAdd := &discordgo.GuildMemberAdd{Member: member}

		events.GuildMemberAddHandler(server)(ctx.Discord, guildMemberAdd)

		return nil
	},
}
